### PR TITLE
chore(docs) Tidying up regexp wording.

### PR DESCRIPTION
**Pull request by amyshwang**

### CODE REVIEW

File: docs/reference/operators/regex.md [hunk 1]
```diff
@@ -5,37 +5,37 @@ description: Regex operators reference documentation.
 ---
 
 This page describes the available operators to assist with performing pattern
-matching via regular expressions.
+matching via regular expressions. QuestDB uses
```
Comment (bziobrowski) [inline docs/reference/operators/regex.md:8]: java.util.regex might not be clear for someone not familiar with Java so how about : 
[java.util.regex implementation] 
or 
[Java regular expression implementation] 
?

File: docs/reference/operators/pattern-matching.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+- `(string) ~ (regex)` - returns true if the `string` value matches the `regex` rules, otherwise returns false (case sensitive match).
+- `(string) !~ (regex)` - returns true if the `string` value fails to match the `regex` rules, otherwise returns false (case sensitive match).
+
+### Arguments
+
+- `string` is an expression that evaluates to the `string` data type.
+- `regex` is any regular expression pattern used to calculate the rules.
+
+## LIKE/ILIKE
+
```
Comment (bziobrowski) [inline docs/reference/operators/pattern-matching.md:22]: case sensitive  vs "case-insensitive" below

File: docs/reference/operators/pattern-matching.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+sidebar_label: Pattern matching
+description: Pattern matching operators reference documentation.
+---
+
+This page describes the available operators to assist with performing pattern
+matching. For operators using regular expressions (`regex` in the syntax), QuestDB uses
+[Java regular expression implementation](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/regex/Pattern.html).
+
+## ~ (match) and !~ (does not match)
+
```
Comment (bziobrowski) [inline docs/reference/operators/pattern-matching.md:12]: case sensitive vs "case-insensitive" below

File: docs/reference/operators/pattern-matching.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+- `(string) ~ (regex)` - returns true if the `string` value matches the `regex` rules, otherwise returns false (case sensitive match).
```
Comment (bziobrowski) [inline docs/reference/operators/pattern-matching.md:13]: I'd double if "matching regex rules" is a valid or popular expression . 
I think normally something simply matches a regular expression .

File: docs/reference/operators/pattern-matching.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+-- Returns true
+
+SELECT 'quest' LIKE 'ques_';   
+-- Returns true   
+
+SELECT 'quest' LIKE 'que%';
+-- Returns true   
+
+SELECT 'quest' LIKE '_ues_';
+-- Returns true
```
[the remark this answers is not in the document]
Comment (amyshwang) [reply docs/reference/operators/pattern-matching.md:56]: Updated - removing empty lines.

File: docs/reference/operators/pattern-matching.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+matching. For operators using regular expressions (`regex` in the syntax), QuestDB uses
+[Java regular expression implementation](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/regex/Pattern.html).
+
+## ~ (match) and !~ (does not match)
+
+- `(string) ~ (regex)` - returns true if the `string` value matches a regular expression, `regex`, otherwise returns false (case sensitive match).
+- `(string) !~ (regex)` - returns true if the `string` value fails to match a regular expression, `regex`, otherwise returns false (case sensitive match).
+
+### Arguments
+
```
Comment (bziobrowski) [inline docs/reference/operators/pattern-matching.md:17]: I'm not sure I understand what 'calculate the rules' means in this context .

File: docs/reference/operators/pattern-matching.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+## LIKE/ILIKE
+
+- `(string) LIKE (pattern)` - returns true if the `string` value matches `pattern`, otherwise returns false (case sensitive match).
+- `(string) ILIKE (pattern)` - returns true if the `string` value matches `pattern`, otherwise returns false (case insensitive match).
+
+### Arguments
+
+- `string` is an expression that evaluates to the `string` data type.
+- `pattern` is a pattern which can contain wildcards like `_` and `%`.
+
```
Comment (bziobrowski) [inline docs/reference/operators/pattern-matching.md:30]: Is this return value block shared by all operators mentioned above ?

Comment (bziobrowski) [reply docs/reference/operators/pattern-matching.md:30]: Might be worth to mention that there's a difference between ~/!~ and like/ilike in that :
- ~ !~ try to match any part of the input string  
- like/like try to match whole string 

Example:

```sql
select 'quest' ~ 'ue'  
--true

select 'quest' like 'ue'  
--false
```

**Review by bziobrowski — APPROVED** [review]

**Pull request merged**